packages/protvista-structure/src/protvista-structure.ts

```diff
@@ -0,0 +1,309 @@
+
+  private _plannedRender: boolean;
+
+  private _selectedMolecule: {
+    id: string;
+    mappings: Mappings;
+  };
+
+  constructor() {
+    super();
+    this._mappings = [];
+
+    this._planHighlight = this._planHighlight.bind(this);
+    this.propagateHighlight = this.propagateHighlight.bind(this);
+  }
+
+  static get is(): string {
+    return "protvista-structure";
+  }
+
+  get css(): string {
+    return `
+    protvista-structure h4 {
+      display: inline;
+      margin-right: 1em;
+    }
+      
+    .structure-viewer-container {
+      position: relative;
+      height: ${this._height};
+    }
+    `;
+  }
+
+  get accession(): string {
+    return this._accession;
+  }
+
+  set accession(accession: string) {
+    this.setAttribute("accession", accession);
+  }
+
+  get pdbId(): string {
+    return this._pdbID;
+  }
+
+  set pdbId(pdbId: string) {
+    this.setAttribute("pdbid", pdbId);
+    this._pdbID = pdbId;
+  }
+
+  get height(): string {
+    return this.getAttribute("height");
+  }
+
+  connectedCallback(): void {
+    const manager = this.closest<HTMLElement>("protvista-manager");
+    if (manager && "register" in manager && "unregister" in manager) {
+      this.manager = manager as NightingaleManager;
+      this.manager.register(this);
+    }
+
+    this._pdbID = this.getAttribute("pdb-id");
+    this._accession = this.getAttribute("accession");
+    this._height = this.getAttribute("height") || "480px";
+    this._highlight =
+      this.getAttribute("highlight") &&
+      ProtvistaStructure._parseHighlight(this.getAttribute("highlight"));
+
+    const style = document.createElement("style");
+    style.innerHTML = this.css;
+    this.appendChild(style);
+
+    const structureViewerDiv = document.createElement("div");
+    structureViewerDiv.className = "structure-viewer-container";
+    structureViewerDiv.id = "structure-viewer-instance";
+    this.appendChild(structureViewerDiv);
+    this._structureViewer = new StructureViewer(
+      structureViewerDiv,
+      this.propagateHighlight
+    );
+  }
+
+  disconnectedCallback(): void {
+    if (this.manager) {
+      this.manager.unregister(this);
+    }
+  }
+
+  static get observedAttributes(): string[] {
+    return ["highlight", "pdb-id", "height"];
+  }
+
+  static _parseHighlight(highlightString: string): HighLight {
+    if (!highlightString) {
+      return [];
+    }
+    const highlightArray = highlightString.split(",").map((region) => {
+      const [_start, _end] = region.split(":");
+      return {
+        start: Number(_start),
+        end: Number(_end),
+      };
+    });
+    return highlightArray;
+  }
+
+  attributeChangedCallback(
+    attrName: string,
+    oldVal: string,
+    newVal: string
+  ): void {
+    if (oldVal !== newVal) {
+      const value = parseFloat(newVal);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore 🤷🏼‍♂️
+      this[`_${attrName}`] = typeof value === "number" ? newVal : value;
```

Comment:
Not sure what this is for as the possible attributes are updated in the code below, and none of them are a number (I think height can be a string because of the units, but if not the `number` check could happen in that block)

Reply:
I can see two attributes from the README that aren't accounted for in the switch statement:

- [hide-table](https://github.com/ebi-webcomponents/nightingale/blob/master/packages/protvista-structure/README.md#hide-table)
- [hide-viewport-controls](https://github.com/ebi-webcomponents/nightingale/blob/master/packages/protvista-structure/README.md#hide-viewport-controls)

I don't think they are implemented anymore - do we want them to be implemented?

Reply:
I don't think so - without them being in the list of observed attributes they won't trigger the `attributChangedCallback` anyway

Reply:
I think these are leftover properties from when Mahdi was trying to add it to PepVep but he ended up doing something else and must have left them in the README by accident

Reply:
👍 removed